### ISSUE TITLE
Training GAN

### ISSUE THREAD

**Issue by NaxAlpha** [pytorch/xla#1547]
## 🐛 Bug

Hi there,

I am trying to train vanilla GAN on single TPU core. I tried both `xrt==1.15` as well as `nightly` package. When using nightly package, run-time crashes but when I use xrt, it gives following error:

![image](https://user-images.githubusercontent.com/11090613/72798220-652c0000-3c64-11ea-960f-a1093edb2145.png)

## To Reproduce

Colab notebook:
https://colab.research.google.com/drive/1buFiGkBuyZtjwNt6u-353QjDyULVUxGR

## Expected behavior

Should be able to train GAN

## Environment

Colab

**Comment by NaxAlpha** [comment]
Yeah I was able to identify the issue. I think it was related to my implementation rather than `xla`. Earlier my kernel was crashing so could not find the bug. 

For future reference, I got it working with `xrt==1.15`

**Comment by dlibenzi** [comment]
Hmm, that would be a bug, as nightly should be at least as stable than 1.15.

**Comment by NaxAlpha** [comment]
Actually I followed the [reference notebook](https://github.com/pytorch/xla/blob/master/contrib/colab/mnist-training-xrt-1-15.ipynb) after my bug got fixed but just checked, it also worked with nightly build.

There were a few differences in behaviour that I noticed so I selected `xrt` version for now. One of the difference that I noticed was `nightly` build took some time while executing the line `device = xm.xla_device()`.

BTW it is amazing to see it working. I have been trying to train GAN on TPU using tf/keras for like months but no success and `pytorch/xla` worked like a magic.
